src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeFormatParser.java

```diff
@@ -1,46 +1,36 @@
 package uk.gov.hmcts.ccd.domain.service.processor;
 
+import java.text.ParsePosition;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
-import java.time.temporal.*;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 import lombok.extern.slf4j.Slf4j;
-import org.joda.time.format.*;
 import org.springframework.stereotype.Component;
 
 @Component
 @Slf4j
 public class DateTimeFormatParser {
 
-    static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS");
-    static final DateTimeFormatter DATE_TIME_FORMAT_MILLISECOND = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SS");
-    static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS";
+    static final String DATE_FORMAT = "yyyy-MM-dd";
     private static final int DEFAULT_YEAR = 1970;
 
     public String convertDateTimeToIso8601(String dateTimeFormat, String value) {
-        LocalDateTime dateTime;
```

Comment:
Don't expose these methods convertDateTimeToIso8601, convertDateToIso8601, instead create a common convert method which accepts baseType (FieldType.DATE, FieldType.DATETIME) and then do the logic based on that.

and Move FieldType.DATE, FieldType.DATETIME to common class